File: backend/src/main/java/com/alab/shinkansendego/traincar/SeatResponseDto.java
package com.alab.shinkansendego.traincar;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SeatResponseDto {
    private FacilityDto frontFacilities;
    private FacilityDto rearFacilities;
    private List<SeatDto> seats;
}

File: frontend/src/features/schedule/types/SeatResponseDto.ts
import type { FacilityDto } from '@/features/schedule/types/FacilityDto';
import type { SeatDto } from '@/features/schedule/types/SeatDto';

export type SeatResponseDto = {
    frontFacilities: FacilityDto;
    rearFacilities: FacilityDto;
    seats: SeatDto[];
};
